src/resolvers/funding.ts

```diff
@@ -230,7 +230,84 @@ export const resolvers: Resolvers = {
       }
     },
 
-    // remove a plan funding
+    // Update any number of planFundings by passing in an array of the new projectFundingIds. New ones will be added and existing ones
+    // that are not included in the projectFundingIds array will be removed
+    updatePlanFunding: async (_, { planId, projectFundingIds }, context: MyContext): Promise<PlanFunding[]> => {
+      const reference = 'updatePlanFunding resolver';
+      try {
+
+        if (!isAuthorized(context.token)) {
+          throw context.token ? ForbiddenError() : AuthenticationError();
+        }
+
+        const plan = await Plan.findById(reference, context, planId);
+        if (isNullOrUndefined(plan)) {
+          throw NotFoundError();
+        }
+
+        const project = await Project.findById(reference, context, plan.projectId);
+        if (await hasPermissionOnProject(context, project)) {
+
+          const associationErrors = [];
+          // Fetch all of the current Funders associated with this Plan
+          const fundings = await PlanFunding.findByPlanId(reference, context, plan.id);
+          const currentPlanFundingids = fundings ? fundings.map((d) => d.projectFundingId) : [];
+          // Use the helper function to determine which funders to keep and which to remove
+          const {
+            idsToBeRemoved,
+            idsToBeSaved
+          } = PlanFunding.reconcileAssociationIds(
+            currentPlanFundingids,
+            projectFundingIds
+          );
+
+          const removeErrors = [];
+          // Remove records that are not in the newly supplied projectFundingIds array
+          for (const id of idsToBeRemoved) {
+            const funding = await PlanFunding.findByProjectFundingId(reference, context, planId, id);
+            if (funding) {
+              const wasRemoved = funding.removeFromPlanFunding(context, planId, id);
+              if (!wasRemoved) {
+                removeErrors.push(funding.projectFundingId);
+              }
+            }
+          }
+          // If any failed to be removed, then add an error
+          if (removeErrors.length > 0) {
+            associationErrors.push(`unable to remove funding: ${removeErrors.join(', ')}`);
+          }
+
+          const addErrors = [];
+          // Add new records for projectFundingIds that are not already in planFundings table
+          for (const id of idsToBeSaved) {
+            const funding = new PlanFunding({ planId, projectFundingId: id });
+
+            const wasAdded = funding.addToPlanFunding(context, planId, id);
+            if (!wasAdded) {
+              addErrors.push(funding.projectFundingId);
+            }
+
+          }
+          // If any failed to be added, then add an error to the ProjectMember
+          if (addErrors.length > 0) {
+            associationErrors.push(`unable to assign roles: ${addErrors.join(', ')}`);
```

Comment:
this is just a copy paste error I think. Should be `unable to add funding: ${addErrors.join(', ')}`